components/script/dom/imagedata.rs

```diff
@@ -27,31 +28,45 @@ pub struct ImageData {
 
 impl ImageData {
     #[allow(unsafe_code)]
-    pub fn new(global: &GlobalScope, width: u32, height: u32, mut data: Option<Vec<u8>>) -> Root<ImageData> {
-        let imagedata = box ImageData {
-            reflector_: Reflector::new(),
-            width: width,
-            height: height,
-            data: Heap::default(),
-        };
-        let len = width * height * 4;
-
+    pub fn from_vec(global: &GlobalScope, width: u32, height: u32, mut data: Option<Vec<u8>>) -> Root<ImageData> {
```

Comment:
The `new` method doesn't validate arguments, neither returns a `Faillible<Root<ImageData>>`. Should it do the same kind of validations as the two new constructors ?